snuba/datasets/storage.py

```diff
@@ -176,9 +176,22 @@ def get_is_write_error_ignorable(self) -> bool:
         return self.__ignore_write_errors
 
 
+class StorageAndMappersNotFound(Exception):
+    pass
+
+
 class StorageAndMappers(NamedTuple):
     storage: ReadableStorage
     mappers: TranslationMappers
+    is_writable: bool
```

Comment:
I think it would be better if `is_writable` wasn't passed in here. This could be calculated based on the storage being passed in. I'm not sure what NamedTuple supports, but some options

- Use the `@property` decorator on a function
- Add a function `is_writable()` that checks if the storage is writable

Reply:
I think that's a better approach too, but there's actually an underlying issue that is blocking the determination of a storage with only the object. The problem is that there a difference between actual storages which inherit from `ReadTableStorage` or `WriteTableStorage` and storages which config defines as "readable" or "writable". For example, see the metrics [sets and counters storages](https://github.com/getsentry/snuba/blob/53d6f3ec05c3e7d306c7697bf6b046c41a4df236/snuba/datasets/storages/metrics.py#L114-L158) which are both `WritableTableStorage`. However, if you look at the [entity](https://github.com/getsentry/snuba/blob/53d6f3ec05c3e7d306c7697bf6b046c41a4df236/snuba/datasets/entities/metrics.py#L127-L164) you'll see that they are "readable" storages on the entity. This is problematic because we use config defined definition to default the storage selector.

Reply:
So if I can rephrase what you're saying, `StorageAndMappers` outline the relationship of an entity to a storage, and just because the storage is `Writeable` doesn't mean the entity sees it that way. Is that right?

Reply:
> just because the storage is Writeable doesn't mean the entity sees it that way
👍 

> StorageAndMappers outline the relationship of an entity to a storage
Sort of. StorageAndMappers outline the relationship of a storage and translation mappers. And we eventually would like an entity to specify a `List[StorageAndMappers]` to therefore outline the relationship between entity and storage.